ngraph/core/shape_inference/include/rnn_cell_shape_inference.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/rnn_cell.hpp>
+
+namespace ov {
+namespace op {
+namespace util {
+template <class T>
+void validate_input_rank(const RNNCellBase* op, const std::vector<T>& input) {
+    enum { X, initial_hidden_state, W, R, B };
+
+    // Verify static ranks for all inputs
+    for (size_t i = 0; i < input.size(); i++) {
+        NODE_VALIDATION_CHECK(dynamic_cast<const ngraph::Node*>(op),
```

Comment:
what is the reasoning for casting?

Reply:
No special reason, just follow original RNN impl. I think it could be removed.

Reply:
resolved in f7d415276b3f8a48e0b32b1bd918a86d70bf6623